src/fasta/record.jl

```diff
@@ -143,38 +136,25 @@ end
 # ------------------
 
 """
-    identifier(record::Record)::Union{StringView, Nothing}
+    identifier(record::Record)::StringView
 
-Get the sequence identifier of `record`.
+Get the sequence identifier of `record`. The identifier is the header
+before any whitespace. If the identifier is missing, return an empty string.
 Returns an `AbstractString` view into the record. If the record is overwritten,
 the string data will be corrupted.
-
-!!! note
-    Returns nothing if the record has no identifier.
 """
-function identifier(record::Record)::Union{String, Nothing}
-    checkfilled(record)
-    if !hasidentifier(record)
-        return nothing
-    end
+function identifier(record::Record)::StringView
     return StringView(view(record.data, record.identifier))
 end
 
-"""
-    hasidentifier(record::Record)
-
-Checks whether or not the `record` has an identifier.
-"""
-function hasidentifier(record)
-    return isfilled(record) && !isempty(record.identifier)
-end
+@deprecate hasidentifier(record::Record) !isempty(identifier(record))
```

Comment:
After these changes, there is always an identifier. It's just that it may be the empty string, but that empty string is then the identifier. So returning `false` would be misleading, I think.

Reply:
I agree with that. But, `hasidentifier` is/was referenced by `BioGenerics.hasseqname` on old-line 177. If `BioGenerics.hasseqname`  is removed, then `hasidentifier` could be removed without deprecation as part of a breaking change.